### PR TITLE
Store an instance to console

### PR DESCRIPTION
**Pull request by polina-c**
RELEASE_NOTE_EXCEPTION=[under flag]

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/panes/diff/controller/sampler.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  HeapClassSampler(this.className);
+
+  final HeapClassName className;
+
+  @override
+  Future<void> oneVariableToConsole() async {
+    final isolateRef = serviceManager.isolateManager.mainIsolate.value!;
+    final isolateId = isolateRef.id!;
+
+    // It would be great to find out how to avoid full scan of classes.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/controller/sampler.dart:21]: should this be a TODO and an issue?

File: packages/devtools_app/lib/src/screens/memory/panes/diff/controller/sampler.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    print('!!!! eval without scope: ' + response1.json!['valueAsString']);
+
+    // eval object
+    final response2 = await serviceManager.service!.evaluate(
+      isolateId,
+      instance.id!,
+      'identityHashCode(this)',
+      scope: {'this': instance.id!},
+    );
+    print('!!!! eval with scope: ' + response2.json!['valueAsString']);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/controller/sampler.dart:57]: are these supposed to still be there? both the responses and the prints?

File: packages/devtools_app/lib/src/screens/memory/panes/diff/controller/heap_diff.dart [hunk 3]
```diff
@@ -111,13 +111,10 @@ class DiffHeapClasses extends HeapClasses<DiffClassStats>
...
   DiffClassStats._({
-    required this.heapClass,
+    required HeapClassName heapClass,
     required this.total,
     required StatsByPath objectsByPath,
-  }) : super(objectsByPath);
-
-  @override
-  final HeapClassName heapClass;
+  }) : super(objectsByPath, heapClass);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/controller/heap_diff.dart:117]: these can both be super parameters in the constructor directly
```
DiffClassStats._({
    required super.heapClass,
    required super.objectsByPath,
    required this.total,
  })
  ```

File: packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_set_view.dart [hunk 4]
```diff
@@ -38,41 +41,46 @@ class InstanceSetView extends StatelessWidget {
...
-      const _MenuForSubset('Store one instance as a console variable'),
-      const _MenuForSubset('Store first 100 instances as a console variable'),
-      const _MenuForSubset('Store all instances as a console variable'),
+List<Widget> _menu(ClassSampler sampleObtainer) => [
+      _StoreAsVariableMenu(
+        'Store as a console variable',
+        sampleObtainer,
+      ),
+      const MenuItemButton(
+        child: Text('Browse an instance in console'),
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_set_view.dart:84]: maybe this should mention the work "references" to make it clear what you can browse

File: packages/devtools_app/lib/src/screens/memory/panes/diff/controller/sampler.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/controller/sampler.dart:57]: they are under TODO to move them out

**Comment by polina-c** [comment]
Addressed the comments in https://github.com/flutter/devtools/pull/5053

File: packages/devtools_app/lib/src/screens/memory/panes/diff/controller/sampler.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/controller/sampler.dart:21]: I am not sure it is possible. So, just TODO to make reader aware.

**Pull request closed without merging**